xarray/coding/variables.py

```diff
@@ -145,11 +146,24 @@ class CFMaskCoder(VariableCoder):
     def encode(self, variable, name=None):
         dims, data, attrs, encoding = unpack_for_encoding(variable)
 
-        if encoding.get('_FillValue') is not None:
+        fv = encoding.get('_FillValue')
+        mv = encoding.get('missing_value')
+
+        if fv is not None and mv is not None and not equivalent(fv, mv):
+            raise ValueError("Variable {!r} has multiple fill values {}. "
+                             "Cannot encode data. "
+                             .format(name, [fv, mv]))
+
+        if fv is not None:
             fill_value = pop_to(encoding, attrs, '_FillValue', name=name)
             if not pd.isnull(fill_value):
                 data = duck_array_ops.fillna(data, fill_value)
 
+        elif mv is not None:
```

Comment:
changed `if` to `elif` so we don't call `fillna` twice if both `_FillValue` and `missing_value` are present

Reply:
Hmm, I think we still want to remove both these keys from `attrs`.